File: 0455-assign-cookies/0455-assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        boolean[] hasEaten = new boolean[g.length];
        int count = 0;
        Arrays.sort(g);
        Arrays.sort(s);
        for (int i = 0; i < s.length; i++) {
            for (int j = 0; j < g.length; j++) {
                if(!hasEaten[j] && g[j] <= s[i]) {
                    hasEaten[j] = true;
                    s[i] = 0;
                    count++;
                }
            }
        }
        return count;
    }
}

File: 0455-assign-cookies/0455-assign-cookies.kt
class Solution {
fun findContentChildren(g: IntArray, s: IntArray): Int {
    g.sort()
    s.sort()
    var i = 0
    var j = 0
    while (i < g.size && j < s.size) {
        if (g[i] <= s[j]) {
            i++
        }
        j++
    }
    return i
}
}